linfa-bayes/src/gaussian_nb.rs

```diff
@@ -0,0 +1,546 @@
+//! Gaussian Naive Bayes (GaussianNB)
+//!
+//! Implements Gaussian Naive Bayes algorithm for classification. The likelihood
+//! of the feature P(x_i | y) is assumed to be Gaussian, the mean and variance will
+//! be estimated using maximum likelihood.
+
+use ndarray::{s, Array1, Array2, ArrayView1, ArrayView2, Axis};
+use ndarray_stats::QuantileExt;
+
+use crate::error::{BayesError, Result};
+use crate::Float;
+
+/// Gaussian Naive Bayes (GaussianNB)
+#[derive(Debug)]
+pub struct GaussianNb<A> {
+    // Prior probabilities of the classes
+    priors: Option<Array1<A>>,
+    // Boolean required for incrementally updating priors when not provided
+    priors_provided: bool,
+    // Required for calculation stability
+    var_smoothing: f64,
+    // Number of training samples observed in each class
+    class_count: Option<Array1<A>>,
+}
+
+impl<A> Default for GaussianNb<A> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<A> GaussianNb<A> {
+    /// Create new GaussianNB model with default values for its parameters
+    pub fn new() -> Self {
+        GaussianNb {
+            priors: None,
+            priors_provided: false,
+            var_smoothing: 1e-9,
+            class_count: None,
+        }
+    }
+
+    // Prior probability of the classes, If set the priors will not be adjusted
+    // according to data
+    pub fn priors(mut self, priors: Array1<A>) -> Self {
+        self.priors = Some(priors);
+        self.priors_provided = true;
+        self
+    }
+
+    // Specifies the portion of the largest variance of all the features that
+    // is added to the variance for calculation stability
+    pub fn var_smoothing(mut self, var_smoothing: f64) -> Self {
+        self.var_smoothing = var_smoothing;
+        self
+    }
+}
+
+impl<A: Float + PartialEq + PartialOrd> GaussianNb<A> {
+    /// Fit the model
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use ndarray::array;
+    /// # use linfa_bayes::GaussianNb;
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// let x = array![
+    ///     [-2., -1.],
+    ///     [-1., -1.],
+    ///     [-1., -2.],
+    ///     [1., 1.],
+    ///     [1., 2.],
+    ///     [2., 1.]
+    /// ];
+    /// let y = array![1., 1., 1., 2., 2., 2.];
+    ///
+    /// let mut clf = GaussianNb::new();
+    /// let fitted_clf = clf.fit(&x, &y)?;
+    /// let pred = fitted_clf.predict(&x)?;
+    ///
+    /// assert_eq!(pred, y);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn fit(&mut self, x: &Array2<A>, y: &Array1<A>) -> Result<FittedGaussianNb<A>> {
```

Comment:
with merging #55 a new fitting API is introduced, this needs an update here

Reply:
Changed